src/sinks/elasticsearch/config.rs

```diff
@@ -307,7 +309,12 @@ impl SinkConfig for ElasticSearchConfig {
     async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
         let common = ElasticSearchCommon::parse_config(self)?;
 
-        let http_client = HttpClient::new(common.tls_settings.clone(), cx.proxy())?;
+        let connection_timeout_duration = Duration::new(self.connection_timeout.unwrap_or(5), 0);
+        let http_client = HttpClient::new(
+            common.tls_settings.clone(),
+            cx.proxy(),
+            Some(connection_timeout_duration),
+        )?;
```

Comment:
Is this unwrap and then `Some` needed? The default timeout is 5 in the core `http.rs` implementation.